### PR TITLE
Set explicit expiration on the HLLs + bonus

### PR DESCRIPTION
**Pull request by spladug**
See commit messages for explanations.

:eyeglasses: @kjoconnor @dellis23

### CODE REVIEW

**Comment by dellis23** [comment]
> One minute causes issue with the activity cron job which runs once a
minute. If the second job is a bit faster than the first, it'll just get
a cached response from the first job.

Can you explain what the "second job" and "first job" are and do?

**Comment by spladug** [comment]
> Can you explain what the "second job" and "first job" are and do?

Sorry, to be more clear it's the first and second runs of the same job.

File: reddit_service_activity/counter.py [hunk 1]
```diff
@@ -43,8 +41,14 @@ def __init__(self, activity_window):
...
-        key = _make_key(context_id)
-        redis.execute_command("PFADD", key, visitor_id)
+        current_slice = _current_slice()
+        key = _make_key(context_id, current_slice)
+        expiration = ((current_slice + 1) * _SLICE_LENGTH) + 1
+
+        with redis.pipeline("record") as pipe:
+            pipe.execute_command("PFADD", key, visitor_id)
+            pipe.expireat(key, expiration)
+            pipe.execute()
```
Comment (dellis23) [inline reddit_service_activity/counter.py]: So previously we were just waiting until we hit some limit and redis automatically evicted, causing long pauses?

File: reddit_service_activity/counter.py [hunk 2]
```diff
@@ -29,9 +29,7 @@ def _current_slice():
     return int(time.time() // _SLICE_LENGTH)
```
Comment (dellis23) [inline reddit_service_activity/counter.py]: The docstring at the top of this file is very helpful!  I should have read it before googling around for HLL stuff.

**Comment by dellis23** [comment]
🐟

File: reddit_service_activity/counter.py (hunk 1 above)
Comment (spladug) [inline reddit_service_activity/counter.py]: Yes

File: reddit_service_activity/counter.py (hunk 2 above)
Comment (spladug) [inline reddit_service_activity/counter.py]: :)

**Pull request merged**